lib/index.js

```diff
@@ -262,6 +262,23 @@ class Study extends EventTarget {
         die();
       }).bind(this)
     )
+    this.on(
+      "uninstall-custom-reason", (function (reason) {
+        prefSvc.get('shield.debug') && console.log("uninstall-custom-reason", this.states);
+        this.states.push("uninstall-custom-reason");
+        if (this.flags.dying) {
+          this.final();
+          return;
+        }
+        this.flags.dying = true;
+        this.report(merge({}, telemetrySubset(this.config), {study_state: reason}), "shield");
```

Comment:
This happened because my code had a `handleHide()` function bound to `panel.onHide` that was trying to to call `button.state` on a `button` object that was removed by `study.shutdown`. So this one was my error.